models/classes/search/SearchProxy.php

```diff
@@ -218,9 +227,7 @@ private function getQueryFactory(): SearchQueryFactory
 
     private function isForcingDefaultSearch(SearchQuery $query): bool
     {
-        $options = $this->getOption(self::OPTION_GENERIS_SEARCH_WHITELIST, []);
-        $generisSearchWhitelist = array_merge(self::GENERIS_SEARCH_DEFAULT_WHITELIST, $options);
-        return in_array($query->getParentClass(), $generisSearchWhitelist, true);
+        return in_array($query->getParentClass(), $this->getGenerisSearchUriWhitelist(), true);
```

Comment:
We need to use $query->getRootClass() to solve issue reported by @AntonNeviarouski 

```suggestion
        return in_array($query->getRootClass(), $this->getGenerisSearchUriWhitelist(), true);
```

Reply:
Fixed here: 841eac5f7079d006ce81b5ad7a9f4b671ef811f2